client-v2/src/shared/util/shared.ts

```diff
@@ -1,17 +1,111 @@
-import { CollectionWithNestedArticles } from 'shared/types/Collection';
+import { CollectionWithNestedArticles, Group } from 'shared/types/Collection';
 import { selectors as collectionSelectors } from 'shared/bundles/collectionsBundle';
 import { selectSharedState } from 'shared/selectors/shared';
 import { State } from 'types/State';
 
 import { normalize, denormalize } from './schema';
+import { CollectionConfig } from 'types/FaciaApi';
+import v4 from 'uuid/v4';
+import keyBy from 'lodash/keyBy';
+
+const createGroup = (
+  id: string,
+  name: string | null,
+  articleFragments: string[] = []
+): Group => ({
+  id,
+  name,
+  uuid: v4(),
+  articleFragments
+});
+
+const getUUID = <T extends { uuid: string }>({ uuid }: T) => uuid;
+
+// this is horrible but is the only way to do it!
+const groupByIndex = (groups: Group[], index: number): Group | undefined =>
+  groups.find(g => parseInt(g.id || '0', 10) === index);
+
+const getAllArticleFragments = (groups: Group[]) =>
+  groups.reduce(
+    (acc, { articleFragments: afs }) => [...acc, ...afs],
```

Comment:
Ah yes, the reason this was there is because there was previously a variable called `articleFragments` in the upper scope which I was shadowing. Happy to replace that back in. Any preferences for what we should call shadowed variables when that _is_ the case?